packages/headless/src/api/commerce/unified-api/unified-api-client.ts

```diff
@@ -0,0 +1,100 @@
+import {Logger} from 'pino';
+import {
+  Badges,
+  Recommendations,
+} from '../../../features/placement-set/placement-set-interface';
+import {PlatformClient} from '../../platform-client';
+import {PreprocessRequest} from '../../preprocess-request';
+import {BadgingRequest} from './badging/badging-request';
+import {RecsRequest} from './recs/recs-request';
+import {
+  UnifiedCommerceAPIError,
+  buildAPIResponseFromErrorOrThrow,
+} from './unified-api-error-response';
+import {
+  baseUnifiedAPIRequest,
+  pickNonBaseUnifiedAPIParams,
+} from './unified-api-params';
+
+export type CommerceUnifiedAPIResponse<TSuccessContent> =
+  | CommerceUnifiedAPISuccessResponse<TSuccessContent>
+  | CommerceUnifiedAPIErrorResponse;
+
+export interface CommerceUnifiedAPISuccessResponse<TContent> {
+  success: TContent;
+}
+
+export interface CommerceUnifiedAPIErrorResponse {
+  error: CommerceUnifiedAPIErrorStatusResponse;
+}
+
+export interface CommerceUnifiedAPIErrorStatusResponse {
+  message: string;
+}
+
+export interface CommerceUnifiedAPIClientOptions {
+  logger: Logger;
+  preprocessRequest: PreprocessRequest;
+}
+
+export type CommerceUnifiedAPIClientResponse<T> =
+  | {success: T}
+  | {error: UnifiedCommerceAPIError};
+
+export class CommerceUnifiedAPIClient {
+  constructor(private options: CommerceUnifiedAPIClientOptions) {}
+
+  async getRecs(
+    req: RecsRequest
+  ): Promise<CommerceUnifiedAPIClientResponse<Recommendations>> {
+    const response = await PlatformClient.call({
+      ...baseUnifiedAPIRequest(req, `/recommendations/${req.placementId}`),
+      method: 'POST',
+      contentType: 'application/json',
+      requestParams: pickNonBaseUnifiedAPIParams(req),
+      ...this.options,
+    });
+
+    if (response instanceof Error) {
+      return buildAPIResponseFromErrorOrThrow(response);
+    }
+    const body = await response.json();
+
+    return response.ok
+      ? {success: body as Recommendations}
+      : {error: body as UnifiedCommerceAPIError};
+  }
+
+  async getBadges(
```

Comment:
Does this endpoint return multiple badges? Or only one?

Reply:
The strict definition of the endpoint is an array of badges.

In practice however, I'm not certain if it's always an array with a single badge 🤔